tests/src/TEST_DataManager.cpp

```diff
@@ -0,0 +1,279 @@
+#include <filesystem>
+#include <fstream>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "DataManager.hpp"
+
+class DataManagerTest : public ::testing::Test {
+protected:
+    struct tablerow {
+        std::string intervention;
+        std::string age;
+        std::string sex;
+        std::string oud;
+        double value;
+    };
+    static int callback(void *storage, int count, char **data, char **columns) {
+        std::vector<struct tablerow> *vecPtr =
+            (std::vector<struct tablerow> *)storage;
+        struct tablerow row;
+        row.intervention = std::string(data[0]);
+        row.age = std::string(data[1]);
+        row.sex = std::string(data[2]);
+        row.oud = std::string(data[3]);
+        row.value = std::stod(std::string(data[4]));
+        vecPtr->push_back(row);
+        return 0;
+    };
+
+    std::filesystem::path tempPath;
+    std::filesystem::path absolute;
+    std::filesystem::path configFile;
+    std::ofstream tempStream;
+    std::ofstream configStream;
+    datamanagement::DataManager testManager;
+    std::string testTableName;
+    void SetUp() override {
+        testManager.ConnectToDatabase("temp.db");
+        tempPath = std::tmpnam(nullptr) + std::string(".csv");
```

Comment:
We shouldn't really be testing on the SCC though right?

Reply:
Perhaps dumb point to raise, but since they use a different flavor of Linux, maybe something would break in differences in implementation between versions of libc and/or glibc? :shrug:

Reply:
That's why docker containers are important. We're testing code functionality not environment compatibility. Integration testing are the next step for sure.